### PR TITLE
Move MapView#setCamera et all to Camera#setCamera

### PR DESCRIPTION
**Pull request by mfazekas**
See: #121
Closes: #64

While camera now has declarative api for moving the camera, i don't think that declarative api is good for this. For example:

1. user presses button to programmatically move the camera to the pisa tower, 
2. user drags the camera to a different location, 
3. user presses the button again to move back to pisa tower. 

Since properties were not changed as user moved camera at (2.) 3. will be a no-op as properties will not be changed. React has controlled input idiom to handle this case, but we don't have it in mapbox, and it's likely non trivial.

### CODE REVIEW

File: docs/Camera.md [hunk 1]
```diff
@@ -22,4 +22,108 @@
...
+| ---- | :--: | :------: | :----------: |
+| `config` | `Object` | `Yes` | Camera configuration |
+
+
+
+```javascript
+this.map.setCamera({
+  centerCoordinate: [lng, lat],
+  zoom: 16,
+  duration: 2000,
```
[the remark this answers is not in the document]
Comment (mfazekas) [reply docs/Camera.md]: I think `animationDuration` should work

[the remark this answers is not in the document]
Comment (kristfal) [reply docs/Camera.md]: AFAIK, there is an issue passing down from the old repo. setCamera doesn’t honor duration on at least iOS. See #1411 in old repo.

File: docs/Camera.md [hunk 2]
```diff
@@ -22,4 +22,108 @@
...
+| Name | Type | Required | Description  |
+| ---- | :--: | :------: | :----------: |
+| `config` | `Object` | `Yes` | Camera configuration |
+
+
+
+```javascript
+this.map.setCamera({
+  centerCoordinate: [lng, lat],
+  zoom: 16,
```
[the remark this answers is not in the document]
Comment (mfazekas) [reply docs/Camera.md]: fixed

File: docs/Camera.md [hunk 3]
```diff
@@ -22,4 +22,108 @@
...
+  duration: 2000,
+})
+
+this.map.setCamera({
```
[the remark this answers is not in the document]
Comment (mfazekas) [reply docs/Camera.md]: thanks, fixed

File: docs/Camera.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfazekas) [reply docs/Camera.md]: `animationDuration` does seems to work - tested on iOS, updated param names and docs.

**Pull request merged**